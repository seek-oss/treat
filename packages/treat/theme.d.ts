type AnyIfEmpty<T extends object> = keyof T extends never ? any : T;

declare module 'treat/theme' {
  export interface Theme {}
  export type ThemeOrAny = AnyIfEmpty<Theme>
}

