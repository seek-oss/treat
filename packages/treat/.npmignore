node_modules
src
tests
tsconfig.json
*.tsbuildinfo