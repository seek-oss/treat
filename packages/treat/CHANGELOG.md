# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.5.1](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.5.0...v1.5.1) (2020-07-27)


### Bug Fixes

* Allow custom treat file module rules ([#128](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/128)) ([a0f1bad](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/a0f1bad))





# [1.5.0](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.4.3...v1.5.0) (2020-07-23)


### Features

* Add hot module reloading support ([#127](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/127)) ([ce5b667](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/ce5b667))





## [1.4.3](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.4.2...v1.4.3) (2020-07-13)


### Bug Fixes

* Ensure consistent and valid content hashes ([#111](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/111)) ([d6993a4](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/d6993a4)), closes [#121](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/121)





## [1.4.1](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.4.0...v1.4.1) (2020-04-26)


### Bug Fixes

* **webpack-plugin:** Checking treat file extension ([#107](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/107)) ([5f5e838](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/5f5e838))





## [1.2.4](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.2.3...v1.2.4) (2020-01-31)


### Bug Fixes

* **styleMap:** Support numbers as keys on 'styleMap' objects ([#88](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/88)) ([1fd5f1f](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/1fd5f1f))





## [1.2.2](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.2.1...v1.2.2) (2020-01-15)


### Bug Fixes

* Detect multiple webpack versions error ([#80](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/80)) ([c067bdf](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/c067bdf))





## [1.2.1](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.2.0...v1.2.1) (2020-01-12)


### Bug Fixes

* Handle undefined watcher error ([#82](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/82)) ([c3b44e9](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/c3b44e9))





## [1.1.7](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.6...v1.1.7) (2019-12-02)


### Bug Fixes

* Produce single type definitions ([#70](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/70)) ([b60aebd](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/b60aebd))





## [1.1.6](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.5...v1.1.6) (2019-10-20)


### Bug Fixes

* Handle zeroes in treat class selectors ([#66](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/66)) ([2d44b72](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/2d44b72))





## [1.1.5](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.4...v1.1.5) (2019-10-15)


### Bug Fixes

* Honour input order for simple pseudos ([#65](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/65)) ([ee7ae3e](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/ee7ae3e))





## [1.1.4](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.3...v1.1.4) (2019-10-08)


### Bug Fixes

* Gracefully handle compilation errors ([#64](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/64)) ([47dd4e8](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/47dd4e8))





## [1.1.3](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.2...v1.1.3) (2019-09-20)


### Bug Fixes

* Normalize CSS file paths for Windows ([#61](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/61)) ([0c96664](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/0c96664))





## [1.1.2](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.1...v1.1.2) (2019-09-20)


### Bug Fixes

* Add theme.d.ts to npm package ([#59](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/59)) ([39a7c63](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/39a7c63))





## [1.1.1](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.1.0...v1.1.1) (2019-09-19)


### Bug Fixes

* Handle tree shaking of themes ([#60](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/60)) ([26ebcd8](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/26ebcd8))





## [1.0.3](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.0.2...v1.0.3) (2019-08-01)


### Bug Fixes

* Fail validation for undefined styles ([#55](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/55)) ([b8399d9](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/b8399d9))





## [1.0.2](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.0.1...v1.0.2) (2019-08-01)


### Bug Fixes

* **deps:** Update Autoprefixer, suppresses browserslist warning ([#54](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/54)) ([dde0281](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/dde0281))





# [1.0.0](https://github.com/seek-oss/treat/tree/master/packages/treat/compare/v1.0.0-beta.2...v1.0.0) (2019-07-24)


### Bug Fixes

* Correctly order themed styles from multiple modules ([#2](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/2)) ([8393fed](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/8393fed))
* Don't publish global theme declaration ([#5](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/5)) ([4e141cc](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/4e141cc))
* Dont throw selector error until all themes found ([#12](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/12)) ([5c01c96](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/5c01c96))
* Handle themed OR selectors in local styles ([#20](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/20)) ([f5e5049](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/f5e5049))
* Make ‘resolveStyles’ support deep objects, add ‘resolveClassName’ ([#6](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/6)) ([a94d920](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/a94d920))
* Mark all treat files as having side effects ([#16](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/16)) ([828a1dd](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/828a1dd))
* **globalStyle:** Handle selectors correctly when no local styles ([#11](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/11)) ([09d80d0](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/09d80d0))
* Mark loader as cacheable ([#23](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/23)) ([53cd2dd](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/53cd2dd))
* Recompile styles when theme dependencies update ([#26](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/26)) ([2f03b8e](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/2f03b8e))
* Remove css dependency when tree shaking ([#4](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/4)) ([114fcf6](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/114fcf6))
* Rename `css` function to `styleMap` ([#7](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/7)) ([110848b](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/110848b))
* **globalStyle:** Correct type signature ([#15](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/15)) ([1b7d025](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/1b7d025))
* **keyframes:** Handle keyframes in selectors block ([#10](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/10)) ([f973d5b](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/f973d5b))
* **resolveClassNames:** Handle theme dereferencing in object keys ([#3](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/3)) ([7957554](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/7957554))
* **selectors:** Enforce targeting of ‘&’ in selectors ([#17](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/17)) ([728c9fa](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/728c9fa))
* **styleTree:** Remove possible styleTree class clashes ([#34](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/34)) ([64862eb](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/64862eb))
* Simplify globalStyles type ([#14](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/14)) ([791fb89](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/791fb89))
* Support Safari < 9 by using var for named exports ([#37](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/37)) ([fc36761](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/fc36761))


### Features

* Add 'styleTree' function ([#28](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/28)) ([88ff843](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/88ff843))
* Add [@keyframes](https://github.com/keyframes) API ([#8](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/8)) ([f115e78](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/f115e78))
* Add runtime validation of styles, optimize compilation ([#18](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/18)) ([5018046](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/5018046))
* Add webpack-plugin option validation ([#32](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/32)) ([a40e7aa](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/a40e7aa))
* Allow ‘undefined’ to be exported from treat files ([#30](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/30)) ([d302abd](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/d302abd))
* Allow themeable global styles ([#19](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/19)) ([1da4b89](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/1da4b89))
* Release v1.0.0 ([#39](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/39)) ([ee7e890](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/ee7e890))


### Performance Improvements

* Add compiler cache ([#25](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/25)) ([616627a](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/616627a))
* Avoid unnecessary rebuilds ([#24](https://github.com/seek-oss/treat/tree/master/packages/treat/issues/24)) ([64c994e](https://github.com/seek-oss/treat/tree/master/packages/treat/commit/64c994e))
