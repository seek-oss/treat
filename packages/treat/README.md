# 🍬 treat

## [Docs](https://seek-oss.github.io/treat/)

See the documentation at seek-oss.github.io/treat for more information about using treat.

## License

MIT.
