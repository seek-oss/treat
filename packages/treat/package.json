{
  "name": "treat",
  "version": "1.6.1",
  "description": "Themeable, statically extracted CSS-in-JS with near-zero runtime",
  "main": "dist/treat.cjs.js",
  "module": "dist/treat.esm.js",
  "browser": {
    "./dist/treat.cjs.js": "./dist/treat.browser.cjs.js",
    "./dist/treat.esm.js": "./dist/treat.browser.esm.js"
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "adapter.ts",
      "webpack-plugin/index.js",
      "webpack-plugin/loader.js"
    ]
  },
  "sideEffects": false,
  "author": "SEEK",
  "license": "MIT",
  "repository": "https://github.com/seek-oss/treat/tree/master/packages/treat",
  "peerDependencies": {
    "webpack": "^4.30.0 || ^5.20.2"
  },
  "dependencies": {
    "@hapi/joi": "^17.1.1",
    "@types/object-hash": "^1.2.0",
    "autoprefixer": "^10.2.4",
    "bluebird": "^3.5.4",
    "chalk": "^4.1.0",
    "css-loader": "^5.0.2",
    "css-selector-parser": "^1.3.0",
    "cssnano": "^4.1.10",
    "csstype": "^3.0.6",
    "debug": "^4.3.1",
    "dedent": "^0.7.0",
    "eval": "^0.1.2",
    "javascript-stringify": "^2.0.0",
    "loader-utils": "^1.2.3",
    "lodash": "^4.17.11",
    "normalize-path": "^3.0.0",
    "object-hash": "^2.0.3",
    "postcss": "^8.2.5",
    "postcss-js": "^3.0.3",
    "style-loader": "^2.0.0",
    "virtual-resource-loader": "^1.0.0"
  },
  "devDependencies": {
    "@types/bent": "^7.3.2",
    "@types/hapi__joi": "^17.1.6",
    "@types/lodash": "^4.14.134",
    "@types/mini-css-extract-plugin": "^1.2.2",
    "bent": "^7.3.12",
    "classnames": "^2.2.6",
    "mini-css-extract-plugin": "^1.3.5",
    "webpack": "^5.20.2"
  }
}
