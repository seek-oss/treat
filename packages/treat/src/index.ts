export {
  createTheme,
  style,
  styleMap,
  styleTree,
  globalStyle,
  globalStyleSheet,
} from './builder';
export { resolveStyles } from './resolveStyles';
export { resolveClassName } from './resolveClassName';

export {
  Style,
  GlobalStyle,
  GlobalStyleSheet,
  CSSProperties,
  ThemeRef,
  ClassRef,
  TreatModule,
} from './types';
