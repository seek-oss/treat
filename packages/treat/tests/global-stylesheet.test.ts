import { globalStyleSheet } from 'treat';

describe('Global stylesheet', () => {
  it('should not accept themed styles within the themed global style sheet', () => {
    expect(() =>
      globalStyleSheet(() => ({
        div: () => {},
      })),
    ).toThrowErrorMatchingSnapshot();
  });

  it('should not accept themed styles within the plain global style sheet', () => {
    expect(() =>
      globalStyleSheet({
        div: () => {},
      }),
    ).toThrowErrorMatchingSnapshot();
  });

  it('should accept multiple unthemed styles', () => {
    expect(() =>
      globalStyleSheet({
        div: {},
        span: {},
      }),
    ).not.toThrow();
  });
});
