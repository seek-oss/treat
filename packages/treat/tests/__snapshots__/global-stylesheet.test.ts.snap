// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global stylesheet should not accept themed styles within the plain global style sheet 1`] = `
"Unthemeable selector: div

Global styles cannot be themed unless the selector references a themed class.

For example, you could add a themed class to the start of your selector (e.g. \`\${themedClass} h1\`).

Without a reference to a themed class, there is no way to selectively apply styles based on the current theme."
`;
